virttest/ppm_utils.py

```diff
@@ -49,7 +49,7 @@ def md5eval(data):
     except NameError:
         hsh = md5.new()
     if data:
-        hsh.update(data)
+        hsh.update(data.encode())
```

Comment:
Seems encoding should be done by callers.

Reply:
I have done the changes also in order to minimize required changes in other parts of code, thus avoiding any breakage. Also, there are other sections of the module that also do encoding at their own initiative.

Reply:
In my point of view, the function `md5eval()` should be a private function because it is not related to `ppm` functionalities, thus, any other codes out of that module should not call it. If that is no doubt, the only left problem is the function `image_md5sum()` of the module called `md5eval()` without encoding the arg `header`, so it just requires an one-line-change as the current approach. The benefit is it does not restrict `md5eval()` to only process strings.

Reply:
I renamed it to private method as well, we can see if this breaks something in the CI testing which is the best we can do here.

Reply:
Cool, thanks.